cmd/azure.go

```diff
@@ -0,0 +1,288 @@
+package cmd
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/storage"
+)
+
+type AzureObjects struct {
+	client storage.BlobStorageClient
+}
+
+func newAzureLayer() (ObjectLayer, error) {
+	creds := serverConfig.GetCredential()
+	c, err := storage.NewBasicClient(creds.AccessKey, creds.SecretKey)
+	if err != nil {
+		return AzureObjects{}, err
+	}
+	client := c.GetBlobService()
+	azure := &AzureObjects{client}
+
+	return azure, nil
+}
+
+func (a AzureObjects) Shutdown() error {
+	return nil
+}
+
+func (a AzureObjects) StorageInfo() StorageInfo {
+	return StorageInfo{}
+}
+
+func (a AzureObjects) MakeBucket(bucket string) error {
+	return traceError(a.client.CreateContainer(bucket, storage.ContainerAccessTypePrivate))
+}
+
+func (a AzureObjects) GetBucketInfo(bucket string) (BucketInfo, error) {
+	resp, err := a.client.ListContainers(storage.ListContainersParameters{
+		Prefix: bucket,
+	})
+	if err != nil {
+		return BucketInfo{}, traceError(err)
+	}
+	for _, container := range resp.Containers {
+		if container.Name == bucket {
+			t, e := time.Parse(time.RFC1123, container.Properties.LastModified)
+			if e != nil {
+				continue
+			}
+			return BucketInfo{
+				Name:    bucket,
+				Created: t,
+			}, nil
+		}
+	}
+	return BucketInfo{}, traceError(BucketNotFound{})
+}
+
+func (a AzureObjects) ListBuckets() (buckets []BucketInfo, err error) {
+	resp, err := a.client.ListContainers(storage.ListContainersParameters{})
+	if err != nil {
+		return nil, traceError(err)
+	}
+	for _, container := range resp.Containers {
+		t, e := time.Parse(time.RFC1123, container.Properties.LastModified)
+		if e != nil {
+			return nil, traceError(e)
```

Comment:
return/continue anything is fine.
but parsing should never fail here, if it does then something very basic has failed and better to return an error else the error gets ignored.